homeassistant/components/waqi/__init__.py

```diff
@@ -1 +1,37 @@
-"""The waqi component."""
+"""The World Air Quality Index (WAQI) integration."""
+from __future__ import annotations
+
+from aiowaqi import WAQIClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+from .coordinator import WAQIDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up World Air Quality Index (WAQI) from a config entry."""
+
+    client = WAQIClient(session=async_get_clientsession(hass))
+    client.authenticate(entry.data[CONF_API_KEY])
```

Comment:
Can this raise or should we do something here if it can't auth?

Reply:
Nope